### PR TITLE
zfs-prune-snapshots: init at 1.1.0

### PR DESCRIPTION
**Pull request by ymarkus**
Also changed ymarkus' email

###### Motivation for this change
Add zfs-prune-snapshots tool

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -9243,7 +9243,7 @@
   };
   ymarkus = {
     name = "Yannick Markus";
-    email = "nixpkgs@ymarkus.dev";
```
Comment (jonringer) [inline maintainers/maintainer-list.nix:9246]: this should be a separate commit:
```
maintainers: update ymarkus email
```

File: pkgs/tools/backup/zfs-prune-snapshots/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchFromGitHub, go-md2man }:
+
+stdenv.mkDerivation rec {
+  version = "1.1.0";
+  pname = "zfs-prune-snapshots";
+  name = "${pname}-${version}";
```
Comment (jonringer) [inline pkgs/tools/backup/zfs-prune-snapshots/default.nix:6]: unneeded
```suggestion
```

File: pkgs/tools/backup/zfs-prune-snapshots/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "09dz9v6m47dxfvfncz0k926dqfhagm87kd33dcw66cbw15ac3spx";
+  };
+
+  nativeBuildInputs = [ go-md2man ];
+
+  buildPhase = ''
+    make man
+  '';
```
Comment (jonringer) [inline pkgs/tools/backup/zfs-prune-snapshots/default.nix:19]:
```suggestion
  makeTargets = [ "man" ];
```

**Comment by ymarkus** [comment]
Thanks, @jonringer. I've changed the lines you marked in the review and tested it. Please re-review!

**Review by jonringer — APPROVED** [review]
LGTM
shows usage

```
https://github.com/NixOS/nixpkgs/pull/94955
1 package built:
zfs-prune-snapshots

```
